Update return value for BeforeInstallEventPrompt.prompt() (#19927)
mdn/content: files/en-us/web/api/beforeinstallpromptevent/prompt/index.md

### Return value

A {{jsxref("Promise")}} resolving to an object containing:

> - `outcome` {{experimental_inline}}  

Reviewer: Actually, this should be `userChoice` (as the spec defines). The compat table should then talk about the non-standard Chrome name. Joe does this in https://github.com/mdn/browser-compat-data/pull/17541#issuecomment-1228726912

```suggestion
- `userChoice` {{experimental_inline}}  
```

Author: But then if we document the spec and not the actual behaviour, the `platform` should be omitted.
Maybe a superior alternative then would be to introduce a new page named `BeforeInstallPromptEvent.AppBannerPromptOutcome` and reference to there instead?

This would of course require adjustments in the BCD so the implementation status there could be documented properly...

Reviewer: Usually non-standard extensions or alternate names appear on the same MDN page.
I think it would be good to give it some more explanation like we do here for moz-opaque: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/canvas#attributes (indicating that there is a non-standard name "outcome" for this but, if possible, to use the standardized name "userChoice" instead.)

For platform, it I think it is fine to leave it there with the indication that it's non-standard.

> - `outcome` {{experimental_inline}}  
>   A string, either `"accepted"` or `"dismissed"`, reflecting the {{jsxref("BeforeInstallPromptEvent.userChoice")}} property.
> 
> - `platform` {{nonstandard_inline}}  
>   A string containing the selected platform, reflecting the {{jsxref("BeforeInstallPromptEvent.platform")}} property.

Reviewer 2: This is incorrect. We use definition lists in that case here. See our [writing guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Markdown_in_MDN#definition_lists).
```suggestion
- `outcome` {{experimental_inline}}  
  - : A string, either `"accepted"` or `"dismissed"`, reflecting the {{jsxref("BeforeInstallPromptEvent.userChoice")}} property.

- `platform` {{nonstandard_inline}}  
  - : A string containing the selected platform, reflecting the {{jsxref("BeforeInstallPromptEvent.platform")}} property.
```

### Return value

A {{jsxref("Promise")}} resolving to an information object containing an `outcome` and a `platform`.  
The `outcome` is either **`"accepted"`** or **`"dismissed"`** (c.f. {{jsxref("BeforeInstallPrompt", "BeforeInstallPrompt.userChoice")}} {{experimental_inline}}).

> The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPrompt", "BeforeInstallPrompt.platform")}} {{experimental_inline}}).

Reviewer: Looking at the example code and at the [old spec](https://www.w3.org/TR/2018/WD-appmanifest-20180220/#dom-promptresponseobject) & [new spec](https://wicg.github.io/manifest-incubations/#dom-promptresponseobject), it seems that `platform` isn't part of the promise object. 

"It contains one member, userChoice, which states the user's chosen outcome."

Author: Considering that I didn't base this on the spec, but what I encountered in chrome (and edge), I'm not surprised that the behaviour mismatches the spec. As mentioned in the PR comment I did not verify the behaviour on any other browsers to see which of them adheres to the spec.

Author: Based off the example already available on the page, running an "abridged"

```js
window.addEventListener("beforeinstallprompt", (e) => {
  e.preventDefault();
  e.prompt().then(r => { console.log('Result of the prompt: ' + r); });
}
```

results in:

```none
Result of the prompt: > {outcome: 'dismissed', platform: ''}
outcome: "dismissed"
platform: ""
[[Prototype]]: Object
```

I am running on Google Chrome version 104.0.5112.101

Reviewer: Thanks, it is strange that it wasn't indicated in the IDL file! I see Joe filed that with Chromium: https://bugs.chromium.org/p/chromium/issues/detail?id=1356264

> The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPrompt", "BeforeInstallPrompt.platform")}} {{experimental_inline}}).

Author: ```diff
The `outcome` is either **`"accepted"`** or **`"dismissed"`** (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.userChoice")}} {{experimental_inline}}).
The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.platform")}} {{experimental_inline}})
```

> The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.platform")}} {{experimental_inline}})

Reviewer: ```suggestion
The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.platform")}} {{nonstandard_inline}})
```

> A {{jsxref("Promise")}} resolving to an information object containing an `outcome` and a `platform`.  
> The `outcome` is either **`"accepted"`** or **`"dismissed"`** (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.userChoice")}} {{experimental_inline}}).
> The `platform` is the selected platform (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.platform")}} {{experimental_inline}})

Reviewer 2: Given that `BeforeInstallPromptEvent.userChoice` is not documented and the link (c.f. {{jsxref("BeforeInstallPromptEvent", "BeforeInstallPromptEvent.userChoice")}} is… unusual.

I would like to suggest:
```suggestion
A {{jsxref("Promise")}} resolving to an information object containing:
- `outcome` that
  - :  A string, either `"accepted"` or `"dismissed"`, reflecting the{{jsxref("BeforeInstallPromptEvent.userChoice")}} property {{experimental_inline}}).
-`platform`
  - : A string containing the selected platform, reflecting the {{jsxref("BeforeInstallPromptEvent.platform")}} property {{nonstandard_inline}}).
```
